e2e/specs/record-food-allergy.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { test } from '../core';
+import { PatientAllergiesPage } from '../pages';
+import { expect } from '@playwright/test';
+import { generateRandomPatient, deletePatient, Patient } from '../commands';
+
+let patient: Patient;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+});
+
+test('Add Food allergy to patient', async ({ page, api }) => {
+  const allergiesPage = new PatientAllergiesPage(page);
+
+  await test.step('When I visit the patient allergies page', async () => {
+    await allergiesPage.goto(patient.uuid);
+  });
+
+  await test.step('And I click the Record allergy intolerance button', async () => {
+    await allergiesPage.page.getByText('Record allergy').click();
+  });
+
+  await test.step('And I click the Food allergen tab', async () => {
+    await allergiesPage.page.getByText('Food').click();
+  });
+
+  await test.step('Then I click the Food allergen option', async () => {
+    await allergiesPage.page.getByText('Eggs').click();
+  });
+
+  await test.step('Then I click the reaction option', async () => {
+    await allergiesPage.page.getByText('Mental status change').click();
+  });
+
+  await test.step('Then I click the severity option', async () => {
+    await allergiesPage.page.getByText('Mild').click();
+  });
+
+  await test.step('Then I fill the comment section', async () => {
+    await allergiesPage.page.locator('#comments').fill('Test comment');
+  });
+
+  await test.step('Then I click the save button', async () => {
+    await allergiesPage.page.getByText('Save').click();
+  });
+
+  await test.step('And I see the Allergy saved message', async () => {
+    await expect(allergiesPage.page.getByText('saved')).toBeVisible();
+  });
+
+  await test.step('And I see the data of the allergy that I saved', async () => {
+    await expect(allergiesPage.tableRow().getByText('Eggs')).toBeVisible();
+    await expect(allergiesPage.tableRow().getByText('Mental status change')).toBeVisible();
+    await expect(allergiesPage.tableRow().getByText('LOW')).toBeVisible();
+    await expect(allergiesPage.tableRow().getByText('Test comment')).toBeVisible();
```

Comment:
When inspecting the cells, it would be better to locate the specific cells instead of checking anywhere within the table. Additionally, verifying the row count is also a good practice.

Consider updating the test code as shown below:

1. For checking the presence of only one record in the allergy table. (The table should not have multiple entries. If so it's a bug.):

```javascript
await test.step('And I see only one record in the allergy table', async () => {
    const rows = await allergyTable.locator('tbody tr');
    await expect(rows.count()).toBe(1);
});

2. For locating and verifying specific cells in the table:
```
```javascript
const allergenCell = await allergyTable.locator('td:first-child');
const severityCell = await allergyTable.locator('td:nth-child(2)');
const thirdCell = await allergyTable.locator('td:nth-child(3)');

await expect(allergenCell).toHaveText('Eggs');
await expect(severityCell).toHaveText('LOW');

```
Please note that I've used `.toHaveText` instead of `.getByText('').toBeVisible()`,


💣 This is just an example, and the actual syntax may differ.

cc: @anjula-sack @Piumal1999

Reply:
@denniskigen, How about adding some examples for common cases in our dev docs? It would be a great idea, right?

Hey @RandilaP, @Piumal1999, @anjula-sack, if anyone's interested in adding a table example to the dev docs, let me know. If not, I'm happy to do it myself. Let's make our docs even better!

Reply:
Thanks @jayasanka-sack for the tips. I'll fix them as soon as possible after I could run my local server normally. Because It's having some issues when It's running

Reply:
@jayasanka-sack @anjula-sack @Piumal1999 I did most of the changes except one. I couldn't do that function that counts the rows. Other changes have been completed. Could you please review it and see if there's something missing?